sonar-plugin-api/src/main/java/org/sonar/api/ce/posttask/QualityGate.java

```diff
@@ -0,0 +1,135 @@
+/*
+ * SonarQube
+ * Copyright (C) 2009-2016 SonarSource SA
+ * mailto:contact AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.api.ce.posttask;
+
+import com.google.common.annotations.Beta;
+import java.util.Collection;
+import javax.annotation.CheckForNull;
+
+/**
+ * @since 5.5
+ */
+@Beta
+public interface QualityGate {
+  String getName();
+
+  /**
+   * Status of the Quality Gate for the current project processing.
+   */
+  Status getStatus();
+
+  /**
+   * Conditions of the Quality Gate.
+   */
+  Collection<Condition> getConditions();
+
+  @Beta
+  enum Status {
+    /** at least one threshold is defined, no threshold is reached */
+    OK,
+    /** at least one warning threshold is reached, no error threshold is reached */
+    WARN,
+    /** at least one error threshold is reached */
+    ERROR
+  }
+
+  @Beta
+  interface Condition {
+    /**
+     * Evaluation status of this condition
+     */
+    EvaluationStatus getStatus();
+
+    /**
+     * The key of the metric this condition has been evaluated on
+     */
+    String getMetricKey();
+
+    /**
+     * The operator used to evaluate the error and/or warning thresholds against the value of the measure
+     */
+    Operator getOperator();
+
+    /**
+     * <p>
+     * At least one of {@link #getErrorThreshold()} and {@link #getWarningThreshold()} is guaranteed to be non {@code null}.
+     * </p>
+     *
+     * @see #getErrorThreshold()
+     */
+    @CheckForNull
+    String getErrorThreshold();
+
+    /**
+     *
+     * <p>
+     * At least one of {@link #getErrorThreshold()} and {@link #getWarningThreshold()} is guaranteed to be non {@code null}.
+     * </p>
+     *
+     * @see #getErrorThreshold()
+     */
+    @CheckForNull
+    String getWarningThreshold();
+
+    /**
+     * Whether this condition is defined on the leak period or on an absolute value
+     */
+    boolean isOnLeakPeriod();
+
+    /**
+     * The value of the measure.
+     *
+     * @throws IllegalStateException if {@link #getStatus()} is {@link EvaluationStatus#NO_VALUE}
+     */
+    String getValue();
```

Comment:
In case of numeric metrics, what is the format of the string value ? Does the decimal separator depend on default locale ?